includes/Core/Authentication/Authentication.php

```diff
@@ -1386,6 +1389,29 @@ private function filter_features_via_proxy( $feature_enabled, $feature_name ) {
 			return $feature_enabled;
 		}
 
+		// The experimental features (ideaHubModule and swgModule) are checked within Modules::construct() which
+		// runs before Modules::register() where the `googlesitekit_features_request_data` filter is registered.
+		// Without this filter, some necessary context data is not sent when a request to Google_Proxy::get_features()
+		// is made. So we skip making this request and solely check the database to see if these features are enabled.
+		if ( in_array( $feature_name, array( 'ideaHubModule', 'swgModule' ), true ) ) {
+			$features = $this->transients->get( $transient_name );
+			if ( false === $features ) {
+				$active_modules = $this->options->get( Modules::OPTION_ACTIVE_MODULES );
+				if ( ! is_array( $active_modules ) ) {
+					$active_modules = $this->options->get( 'googlesitekit-active-modules' );
+				}
```

Comment:
Good catch to include this here, but I think we can get rid of it to have less duplicate code here, since when this old option was being used, neither `ideaHubModule` nor `swgModule` existed anyways.

I'm not even opposed to removing the same line from the `Modules` class itself maybe, but especially here we don't need to support it.